src/nl/rubensten/texifyidea/window/ImagePanel.kt

```diff
@@ -0,0 +1,71 @@
+package nl.rubensten.texifyidea.window
+
+
+import javax.swing.*
+import java.awt.*
+import java.awt.image.BufferedImage
+import java.lang.Integer.max
+import java.lang.Double.min
+
+internal class ImagePanel : JPanel() {
+
+    private var img: Image? = null
+    private var img_width: Int = 1
+    private var img_height: Int = 1
+    private var scaled: Image? = null
+
+    fun set_image(img: BufferedImage) {
+        this.img = img
+        this.img_height = img.height
+        this.img_width = img.width
+        this.invalidate()
+        revalidate()
+        repaint()
+    }
+
+    fun clear_image() {
+        this.img = null
+        this.scaled = null
+        this.img_height = 1
+        this.img_width = 1
+        this.invalidate()
+        revalidate()
+        repaint()
+    }
+
+    override fun invalidate() {
+        super.invalidate()
+
+        if (img == null) return
+
+        val width = max(50,width)
+        val height = max(50,height)
+
+        val ratio = min(width.toDouble()/img_width,height.toDouble()/img_height)
+
+        val w = (img_width*ratio).toInt()
+        val h = (img_height*ratio).toInt()
+
+        scaled = img?.getScaledInstance(w, h, Image.SCALE_SMOOTH)
+    }
+
+    override fun getPreferredSize(): Dimension {
+        return if (img == null)
+            Dimension(200, 200)
+        else
+            Dimension(
+                    img!!.getWidth(this), img!!.getHeight(this))
```

Comment:
Add braces to the first if. The last else may have none, but only when put on the same line. Also, we wrap at 100 characters :) so you can extend the last line.

```kotlin
return  if (image == null) {
   Dimension(200, 200)
}
else Dimension(img!!.getWidth(this), img!!.getHeight(this))
```